x-pack/solutions/search/plugins/search_connectors/server/services/index.ts

```diff
@@ -25,10 +29,15 @@ export interface PackageConnectorSettings {
   service_type: string;
 }
 
+export type AgentMetadata = Pick<Agent, 'last_checkin_status' | 'id' | 'status'>;
+
 export interface PackagePolicyMetadata {
   package_policy_id: string;
+  package_policy_name: string;
+  package_name: string;
   agent_policy_ids: string[];
   connector_settings: PackageConnectorSettings;
+  agent_metadata?: AgentMetadata;
```

Comment:
We should document why it's not available in other methods.

Ideally it should be:
1. Always populated
2. A different object that outlines what is returned so that it's more obvious why and what is returned

If we keep adding nullable fields that are populated conditionally, it quickly becomes a large problem.

Reply:
you are right 👍 I will create a new tsx type for this method the returns agent metadata

Reply:
Addressed in https://github.com/elastic/kibana/pull/210222/commits/1897dacd79d7b8104a7a203ade9c09566ef0d4f8